cartographer/io/image_utils.cc

```diff
@@ -0,0 +1,67 @@
+#include "cartographer/io/image_utils.h"
+
+#include <memory>
+
+#include "cairo/cairo.h"
+
+#include "glog/logging.h"
+
+#include "cartographer/io/file_writer.h"
+
+namespace cartographer {
+namespace io {
+namespace {
+
+// std::unique_ptr for Cairo surfaces. The surface is destroyed when the
+// std::unique_ptr is reset or destroyed.
+using UniqueSurfacePtr =
+    std::unique_ptr<cairo_surface_t, void (*)(cairo_surface_t*)>;
+
+cairo_status_t CairoWriteCallback(void* const closure,
+                                  const unsigned char* data,
+                                  const unsigned int length) {
+  if (static_cast<FileWriter*>(closure)->Write(
+        reinterpret_cast<const char*>(data), length)) {
+    return CAIRO_STATUS_SUCCESS;
+  }
+  return CAIRO_STATUS_WRITE_ERROR;
+}
+
+constexpr cairo_format_t kCairoFormat = CAIRO_FORMAT_ARGB32;
+
+int stride_for_width(int width)
+{
+  const int stride = cairo_format_stride_for_width(kCairoFormat, width);
+  CHECK_EQ(stride % 4, 0);
+  return stride;
+}
+
+}  // namespace
+
+Image CreateImage(int width, int height)
```

Comment:
It feels like this should now be a class `Image` with a constructor. You can add pixel accessors and hide all of the Cairo implementation in this file.